security/self-assessment.md

```diff
@@ -0,0 +1,363 @@
+management of bare metal machines. It is known as the baremetal service for
+OpenStack and it is a powerful tool on its own, adding ways to be deployed
+independently as a standalone service, for example using Bifrost, and integrates
+in other tools and projects, as in the case of Metal3. As mentioned above,
+Metal³ handles the installation of Ironic as a standalone component. Bare Metal
+Operator is the main component that interfaces with the Ironic API for all
+operations needed to provision bare-metal hosts, such as hardware capabilities
+inspection, operating system installation, and re-initialization when restoring
+a bare-metal machine to its original status. Metal3 provides a way to install
+Ironic with a suitable configuration. Currently Metal3 is in the process of
+introducing a stand alone operator to deploy ironic instances.
+
+Alternatively, Bare Metal Operator can be set up to use an externally managed
+Ironic instance.
+
+### Actions
+
+Metal3 follows the Kubernetes declarative model where users interact with the
+system by creating, updating and deleting Kubernetes custom resources. The main
+actions flow through several key components:
+
+1. Users submit Kubernetes manifests (CRs) to define desired state for baremetal
+   hosts and clusters
+1. CAPM3 (Cluster API Provider Metal3) receives the desired state of the
+   cluster, the machine specs and infrastructure configurations. It sets the
+   infrastructure components accordingly, interacts with IPAM for static IP
+   allocation and also communicates the desired state of the metal3 machine and
+   chooses a baremetal host to be consumed accordingly.
+1. Baremetal Operator validates and processes BareMetalHost resources,
+   performing security checks on credentials and configurations
+1. BMO interacts with Ironic using authenticated API calls to execute the actual
+   provisioning operations
+1. IPAM manages IP address allocation requests securely through IPAddress custom
+   resources
+1. All component interactions happen through the Kubernetes API server,
+   inheriting its authentication and authorization controls
+
+The security boundary between components is maintained through:
+
+* Kubernetes RBAC controls for API access
+* Authentication between components
+* Mutual TLS configuration
+* Validation of all inputs before processing
+* Secure storage of sensitive data like BMC credentials in Kubernetes secrets
+
+### Goals
+
+There are a number of great open source tools for bare metal host provisioning,
+including Ironic. Metal3 aims to build on these technologies to provide a
+Kubernetes native API for managing bare metal hosts via a provisioning stack
+that is also running on Kubernetes. We believe that Kubernetes Native
+Infrastructure, or managing your infrastructure just like your applications, is
+a powerful next step in the evolution of infrastructure management.
+
+The Metal3 project is also building integration with the Kubernetes cluster-api
```

Comment:
"building integration" sounds like we don't have it fully working yet to me
```suggestion
The Metal3 project is also an infrastructure provider with the Kubernetes cluster-api
```

Reply:
Agree, it can be understood differently. I'll address that.

Reply:
Forgotten?